NEventSocket/FreeSwitch/EventMessage.cs

```diff
@@ -71,13 +84,16 @@ internal EventMessage(BasicMessage basicMessage)
                 {
                     // ...but some Event Messages also carry a body payload, eg. a BACKGROUND_JOB event
                     // which is a message carried inside an EventMessage carried inside a BasicMessage..
-                    Headers = basicMessage.BodyText.Substring(0, delimiterIndex).ParseKeyValuePairs(": ");
+                    string headersSection = Encoding.UTF8.GetString(basicMessage.BodyBytes, 0, delimiterIndex);
+                    Headers = headersSection.ParseKeyValuePairs(": ");
 
                     Debug.Assert(Headers.ContainsKey(HeaderNames.ContentLength));
                     var contentLength = int.Parse(Headers[HeaderNames.ContentLength]);
 
-                    Debug.Assert(delimiterIndex + 2 + contentLength <= basicMessage.BodyText.Length, "Message cut off mid-transmission");
-                    var body = basicMessage.BodyText.Substring(delimiterIndex + 2, contentLength);
+                    Debug.Assert(Headers.ContainsKey(HeaderNames.EventName));
+
+                    Debug.Assert(delimiterIndex + 2 + contentLength <= basicMessage.BodyBytes.Length, "Message cut off mid-transmission");
+                    var body = Encoding.UTF8.GetString(basicMessage.BodyBytes, delimiterIndex + 2, contentLength);
 
                     //remove any \n\n if any
                     var index = body.IndexOf("\n\n", StringComparison.Ordinal);
```

Comment:
This does not remove `\n\n`, it truncates the Body before `\n\n`. What is the intention here? Because from my understanding, the body should be treated as opaque.

@ajgolledge We should be able to create a test case, for example, a speech detection result could _legally_ contain a `\n\n` at which it would be truncated.